src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,151 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { addComment } from '../services/comment';
+
+type Props = {
+  selectedPost: Post;
+  setComments: (value: Comment[] | { (prev: Comment[]): Comment[] }) => void,
+  setErrorMessage: (value: string) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setComments,
+  setErrorMessage,
+}) => {
+  const initialFormValues = {
+    name: '',
+    email: '',
+    comment: '',
+  };
+
+  const initialErrors = {
+    nameError: false,
+    emailErrorMessage: '',
+    commentError: false,
+  };
+
+  const [formValues, setFormValues] = useState(initialFormValues);
+  const [errors, setErrors] = useState(initialErrors);
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setErrors({
+      ...errors,
+      nameError: false,
+    });
+    setFormValues({
+      ...formValues,
+      name: event.target.value,
+    });
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setErrors({
+      ...errors,
+      emailErrorMessage: '',
+    });
+
+    setFormValues({
+      ...formValues,
+      email: event.target.value,
+    });
+  };
+
+  const handleCommentChange = (
+    event: React.ChangeEvent<HTMLTextAreaElement>,
+  ) => {
+    setErrors({
+      ...errors,
```

Comment:
you can create only one handle function 
```
const handleChange = (e: ...) => {
  const {name, value} = e.target;
  setFormValues((prev) => ({...prev, [name]: value}))
  setErrors((prev) => ({...prev, [name]: value}))
  
}
```

don't forget to specify name attr to inputs and change errors state